exporter/kafkaexporter/pdata_marshaler.go

```diff
@@ -14,30 +14,68 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatautil"
 )
 
+// KeyableLogsMarshaler is an extension of the LgosMarshaler interface intended to provide partition key capabilities
+// for log messages
+type KeyableLogsMarshaler interface {
```

Comment:
Does it need to be exported?

Reply:
I'm not familiar enough to answer this, but a default of not exposing unless there's a reasons seems reasonable 😄 Since the others (metrics, traces) _are_ currently public, I'm not going to change those but I'm happy to change this.

Reply:
Changed this new type in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33633/commits/71e47515b76bd1531e97be99d475bbd3fe17b315